simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/evse/Connector.java

```diff
@@ -66,7 +58,7 @@ public Integer unplug() {
      *
      * @return identity of the connector
      */
-    public Integer lock() {
+    public synchronized Integer lock() {
```

Comment:
It is called from Evse#lockPluggedConnector. If only this part is synchronized - it will be possible to lock connector with unavailable Evse